usagov/usagov-2021
# USAGOV-2014: Add "data-" attributes to divs to the allowed Source Editing tags in the USWDS editor config

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-2014

## Description

Adds a small module that removes a restriction in ckeditor5.plugin.ckeditor5_sourceEditing, and adds "div data-*" to the elements allowed in "Source Editing" mode for the USWDS editor configuration. 

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions



 - Make sure to import configuration ("drush cim") 
 - Edit the body of a page like /phone . Select the USWDS editor, and then click the Source Editing button. 
 - Add a div with a data attribute, like say "data-call-wait='showTime'"
 - Save

Your div should show up on the rendered page with its data attribute intact. 

You can also inspect the editor configuration by going to http://localhost/admin/config/content/formats and clicking "configure" by USWDS. Click "Source editing" to see that "div data-*" is in the list. (It's also shown in "Limit allowed HTML tags and correct faulty HTML" under "Filter settings," along with many other allowed tags.)

### Change Requirements



### Validation Steps

- Test instruction 1
- Test instruction 2
- Test instruction 3

## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
